flink-table/flink-table-planner/src/main/java/org/apache/flink/table/operations/AlgebraicOperationFactory.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.operations;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.table.api.TableException;
+import org.apache.flink.table.api.TableSchema;
+import org.apache.flink.table.api.ValidationException;
+import org.apache.flink.table.plan.logical.Intersect;
+import org.apache.flink.table.plan.logical.LogicalNode;
+import org.apache.flink.table.plan.logical.Minus;
+import org.apache.flink.table.plan.logical.Union;
+
+import java.util.stream.IntStream;
+
+import static org.apache.flink.table.operations.AlgebraicOperationFactory.AlgebraicTableOperationType.UNION;
+
+/**
+ * Utility class for creating a valid algebraic operation such as {@link Minus}, {@link Intersect} or {@link Union}.
+ */
+@Internal
+public class AlgebraicOperationFactory {
```

Comment:
Call this `SetOperation` instead (also in comments and related classes) to be consistent with tests, rules, etc.

https://en.wikipedia.org/wiki/Set_operations_(SQL)